homeassistant/components/bluesound/media_player.py

```diff
@@ -831,6 +842,41 @@ def shuffle(self):
         else:
             _LOGGER.error("Master not found %s", master_device)
 
+    @property
+    def device_state_attributes(self):
+        """List members in group."""
+        attributes = {}
+        if self._group_list != []:
+            attributes = {ATTR_BLUESOUND_GROUP: self._group_list}
+
+        attributes[ATTR_MASTER] = self._is_master
+
+        return attributes
+
+    def rebuild_bluesound_group(self):
+        """Rebuild the list of entities in speaker group."""
+        if self._group_name is None:
+            return None
+
+        device_group = []
+        bluesound_group = []
+
+        device_group = self._group_name.split("+")
+        for entity in self._hass.data[DATA_BLUESOUND]:
+            if (
+                entity.bluesound_device_name in device_group
+                and entity.is_master is True
+            ):
+                bluesound_group.append(entity.name)
+        for entity in self._hass.data[DATA_BLUESOUND]:
```

Comment:
Instead of iterating twice to get the correct order, use `sorted` to sort the entities on `is_master` key:
https://docs.python.org/3/library/functions.html#sorted

Something like this:
```py
sorted_entities = sorted(
    self._hass.data[DATA_BLUESOUND],
    key=lambda entity: entity.is_master
)
bluesound_group = [
    entity.name for entity in sorted_entities
    if entity.bluesound_device_name in device_group
]
```

Reply:
Nice, much cleaner like this... thanks... Changed it and committed the changes.